shared/src/validation-schemas/entity-id-parameters.validation-schema.ts

```diff
@@ -0,0 +1,12 @@
+import { z } from 'zod';
+
+import { ParametersValidationMessage } from '../enums/enums.js';
+
+const idParameterSchema = z.object({
+    id: z
+        .string()
+        .regex(/^\d+$/, ParametersValidationMessage.INVALID_ID)
+        .transform(Number),
+});
```

Comment:
```suggestion
const idParameterSchema = z.object({
    id: z.coerce.number().min(1).int()
});
```
Wouldn't this be better?